### DISCUSSION TITLE
Special Phrases

### DISCUSSION THREAD

**Discussion by BobLamarley** [mediagis/nominatim-docker#586] [General]
Hello guys ( and girls),
I'm trying to install special phrases as i found in the nominatim documentation : https://nominatim.org/release-docs/4.4/customize/Special-Phrases/
But how to activate it when lauching the docker command :
```
docker run -it \
  -e PBF_URL=https://download.geofabrik.de/europe/monaco-latest.osm.pbf \
  -e REPLICATION_URL=https://download.geofabrik.de/europe/monaco-updates/ \
  -p 8080:8080 \
  --name nominatim \
  mediagis/nominatim:4.4
```
 ?
 
 Also if i want to use overpass, it's another service ? Is there a docker image for start this service and connect it also with nominatim nstance who's running ?

**Comment by mtmail** [comment]
The first "nominatim" is the user name, the second "nominatim" is the name of running Docker container

`docker exec -it nominatim sudo -u nominatim nominatim special-phrases --import-from-wiki`

**Reply by BobLamarley** [reply]
I've tried to import special phrases from inside the container but it seems to have permission problem:
<img width="1088" alt="image" src="https://github.com/user-attachments/assets/b89ba4b2-51d3-4379-8b0b-fd26364d7eb4">
The dir /app/tokenizer doesn't exist, and i can't create it when using nominatim user, maybe you have an idea why ?
Thanks

**Reply by mtmail** [reply]
> Using project directory: /app

I believe the project directory is `/nominatim`, at least according to the `Dockerfile` and `start.sh`. Try to change into that directory first.

**Reply by BobLamarley** [reply]
Yeah you was right thanks it's much better with : `docker exec -it nominatim bash -c "cd /nominatim && sudo -u nominatim nominatim special-phrases --import-from-wiki"`
It add all special phrases for all languages :
<img width="870" alt="image" src="https://github.com/user-attachments/assets/10079cf7-7032-45a9-b5c3-84222c4b476f">
I think FR and EN will be enough in my case, maybe i can limit to them ?
